build/nlu-regression.js

```diff
@@ -0,0 +1,218 @@
+var http = require('http')
+var fs = require('fs')
+const { exit } = require('process')
+
+const repoRootDir = `${__dirname}/..`
+const nluTestingDir = `${repoRootDir}/modules/nlu-testing/`
+
+async function post(path, content, headers) {
+  var post_options = {
+    host: '127.0.0.1',
+    port: '3000',
+    path,
+    method: 'POST',
+    headers,
+    timeout: 2000
+  }
+
+  return new Promise(function(resolve, reject) {
+    var post_req = http.request(post_options, function(res) {
+      res.setEncoding('utf8')
+
+      let receivedData = ''
+      res.on('data', function(chunk) {
+        receivedData += chunk
+      })
+      res.on('error', function(err) {
+        reject(err)
+      })
+      res.on('end', function() {
+        resolve(receivedData)
+      })
+    })
+
+    // post the data
+    post_req.write(content)
+    post_req.end()
+  })
+}
+
+async function get(path, headers) {
+  var post_options = {
+    host: '127.0.0.1',
+    port: '3000',
+    path,
+    method: 'GET',
+    headers
+  }
+
+  return new Promise(function(resolve, reject) {
+    var get_req = http.request(post_options, function(res) {
+      res.setEncoding('utf8')
+
+      let receivedData = ''
+      res.on('data', function(chunk) {
+        receivedData += chunk
+      })
+      res.on('error', function(err) {
+        reject(err)
+      })
+      res.on('end', function() {
+        resolve(receivedData)
+      })
+    })
+    get_req.end()
+  })
+}
+
+async function login() {
+  var data = 'email=admin&password=123456'
+  const rawLogin = await post('/api/v1/auth/login/basic/default', data, {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(data)
+  })
+  const login = JSON.parse(rawLogin)
+
+  const token = login.payload && login.payload.token
+  return login.statusCode ? undefined : token
+}
+
+async function signup() {
+  var data = 'email=admin&password=123456'
+  const rawLogin = await post('/api/v1/auth/register/basic/default', data, {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(data)
+  })
+  const login = JSON.parse(rawLogin)
+
+  const token = login.payload && login.payload.token
+  return login.statusCode ? undefined : token
+}
```

Comment:
This would be clearer. No need of all the validations, if the payload doesn't contain the token, any step failing will return simply undefined

```js
try {
  const rawLogin
  return JSON.parse(rawLogin).payload.token
} catch(err){
  return undefined
}
```